### PR TITLE
fix: UIUserInterfaceStyle is undefined for iOS <= 11

### PR DESCRIPTION
**Pull request by vchimev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [x] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

Fix that UIUserInterfaceStyle is undefined for iOS <= 11.
Add tests for the system appearance property and system appearance root view CSS classes.

Relates to https://github.com/NativeScript/NativeScript/issues/7800.

### CODE REVIEW

File: tns-core-modules/application/application.d.ts [hunk 1]
```diff
@@ -452,6 +452,7 @@ export class AndroidApplication extends Observable {
     /**
      * Gets the system appearance.
      * Available values: "dark", "light".
+     * Undefined for iOS <= 11.
```
Comment (MartoYankov) [inline tns-core-modules/application/application.d.ts]: You should include this in the return types.

**Comment by MartoYankov** [comment]
The property `userInterfaceStyle` doesn't exist on `UITraitCollection` before iOS 12. This isn't obvious in the code. I think it would be more obvious if you check for major version at the beginning of the `systemAppearance` getter and if < 12 return `light` or `undefined` explicitly.

**Comment by SvetoslavTsenov** [comment]
test --ignore cuteness uitests#vchimev/dark-mode/fixes  ios10 ios11 ios12 ios13

**Comment by vchimev** [comment]
test --ignore cuteness uitests#vchimev/dark-mode/fixes ios10 ios11 ios12 ios13

**Comment by dtopuzov** [comment]
test --ignore cuteness uitests#vchimev/dark-mode/fixes ios10 ios11 ios12 ios13

**Comment by dtopuzov** [comment]
test --ignore cuteness uitests#vchimev/dark-mode/fixes ios10 ios11 ios12 ios13

**Comment by manoldonev** [comment]
test --ignore cuteness uitests#vchimev/dark-mode/fixes ios10 ios11 ios12 ios13

**Pull request merged**